### DISCUSSION TITLE
Be able to apply discounts only on specific Sales channels and POS location/s

### DISCUSSION THREAD

**Discussion by pablogiralt** [Shopify/function-examples#270] [Feedback: Discounts APIs]
Add Sales Channel and POS location as function's inputs so discounts can be applied:
- Only to Online Store, Only in POS or in both
- When applied on POS, be able to apply only in certain POS locations.

Our use case is a food merchant which has different discounts in different shops based on the expiration date of the produce.

Please upvote and add your use case if you find this useful

**Comment by realpratiknikam** [comment]
This would def give some flexibility!

**Comment by chesterwu** [comment]
Same here! +1

**Comment by meandillar** [comment]
Also need this for Shipping Customizations to allow for hiding/showing shipping rates for Online Store vs POS

**Comment by thosakwe** [comment]
It would be great if  `Cart` object had a `salesChannel` field. This way, Functions could check whether an order were on Online Store vs. POS, etc.

**Reply by joeainsworth** [reply]
This would be awesome!

**Comment by joeainsworth** [comment]
The way I'm approaching this;
Create a theme app extension which adds an order attribute with a value of online sales channel
Apply discount if this attribute is present or missing (allows you to target online or POS or both)

**Comment by Behnam-RK** [comment]
From `2025-07` version, Shopify discount function APIs have a new `retailLocation` Input that can be used to distinguish POS carts!
More at: https://shopify.dev/changelog/pos-specific-shopify-function-logic
